h2o-algos/src/test/java/hex/DTMatrix/DMatrixTest.java

```diff
@@ -0,0 +1,276 @@
+package hex.DTMatrix;
+
+/**
+ * Created by wendycwong on 1/6/17.
+ */
+
+
+import hex.DMatrix;
+import hex.DataInfo;
+import hex.pca.PCA;
+import hex.pca.PCAModel;
+import hex.pca.PCAModel.PCAParameters;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import water.Key;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.util.ArrayUtils;
+
+import java.util.concurrent.ExecutionException;
+
+import static java.lang.Math.abs;
+
+public class DMatrixTest extends TestUtil {
+    public static final double TOLERANCE = 1e-6;
+    @BeforeClass public static void setup() { stall_till_cloudsize(1); }
+
+    /*
+  This unit test uses the prostate datasets with NAs, calculate the eigenvectors/values with original PCA implementation.
+  Next, it calculates the eigenvectors/values using PCA with wide dataset flag set to true.  Then, we
+  compare the eigenvalues/vectors from both methods and they should agree.  Dataset contains numerical and
+  categorical columns.
+   */
+    @Test public void testWideDataSetsWithNAs() throws InterruptedException, ExecutionException {
+        PCAModel modelN = null;     // store PCA models generated with original implementation
+        PCAModel modelW = null;     // store PCA models generated with wideDataSet set to true
+        Frame train = null, scoreN = null, scoreW = null;
+        try {
+            train = parse_test_file(Key.make("prostate_cat.hex"), "smalldata/prostate/prostate_cat.csv");
+            train.vec(0).setNA(0);
+            train.vec(3).setNA(10);
+            train.vec(5).setNA(100);
+            PCAModel.PCAParameters parms = new PCAModel.PCAParameters();
+            parms._train = train._key;
+            parms._k = 7;
+            parms._transform = DataInfo.TransformType.STANDARDIZE;
+            parms._use_all_factor_levels = true;
+            parms._pca_method = PCAParameters.Method.GramSVD;
+            parms._impute_missing=false;
+            parms._seed = 12345;
+
+            PCA pcaParms = new PCA(parms);
+            modelN = pcaParms.trainModel().get(); // get normal data
+            scoreN = modelN.score(train);
+
+            PCA pcaParmsW = new PCA(parms);
+            pcaParmsW.setWideDataset(true);  // force to treat dataset as wide even though it is not.
+            modelW = pcaParmsW.trainModel().get();
+            scoreW = modelW.score(train);
+
+            // check to make sure eigenvalues and eigenvectors are the same
+            // compare eigenvectors and eigenvalues generated by original PCA and wide dataset PCA.
+            TestUtil.checkStddev(modelW._output._std_deviation, modelN._output._std_deviation, 1e-2);
```

Comment:
1e-2 seems high tolerance, especially since without nas you have 1e-12. Seems suspicious.